services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/MasterIndexFileCreator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Corona-Warn-App
+ *
+ * SAP SE and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ *  file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package app.coronawarn.server.services.distribution.assembly.component;
+
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ * This service will create the master index file for all hourly package files present on this distribution.
+ */
+@Service
+public class MasterIndexFileCreator {
+
+  private static final Logger logger = LoggerFactory.getLogger(MasterIndexFileCreator.class);
+
+  private static final Pattern HOUR_INDEX_PATTERN = Pattern.compile(".*/hour/\\d{1,2}/index$");
```

Comment:
No patterns, parsing or directory scanning required when this is a decorator. Then you can simply use the already provided functionality of [Directory](https://github.com/corona-warn-app/cwa-server/blob/master/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/directory/Directory.java) to achieve this. Example what this could look like can be found [here](https://github.com/corona-warn-app/cwa-server/blob/master/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/directory/decorator/DateAggregatingDecorator.java). Also consider using [getWritablesInDirectory](https://github.com/corona-warn-app/cwa-server/blob/master/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/directory/decorator/DateAggregatingDecorator.java#L123).